players/base.py

```diff
@@ -359,7 +359,9 @@ def escaped(self) -> bool:
     
     def is_defending(self) -> bool:
         defend_byte = self._read_char_battle_state_address(offset=PlayerMagicNumbers.DEFENDING)
-        return (defend_byte >> 3) == 1
+        result = (defend_byte >> 3) & 1 == 1
```

Comment:
This fix should probably be pulled out separately from doing RNG manipulation that we likely don't want to merge in. I would suggest making a new PR with just this change in it.

Reply:
sounds good. I found some other problems too, working through those.